cmd/broker/filter/main.go

```diff
@@ -146,6 +148,19 @@ func main() {
 	logger.Info("Exiting...")
 }
 
+// Same as metrics.UpdateExporterFromConfigMap except the PrometheusPort.
+func updateExporterFromConfigMap(component string, logger *zap.SugaredLogger) func(configMap *corev1.ConfigMap) {
```

Comment:
Would it make sense to change metrics.UpdateExporterFromConfigMap signature to something like:
UpdateExporterFromConfigMap(component string, logger *zap.SugaredLogger, port ...int) {
   and then in the function
if len(port) > 0
wire it in there?

Only reason for varargs style is to keep the old signature and not have to change all the callers at once, but just a thought.

Reply:
This does seem useful as more knative components might use the metrics package but also want to avoid conflict with queue-proxy. I will make the upstream change.

Reply:
> Only reason for varargs style is to keep the old signature and not have to change all the callers at once, but just a thought.

As long as we eventually make it a single arg once every caller has transitioned. Might be easier to just change everything now, early in the release cycle.

Reply:
Yesh, I just wanted to provide an alternative in case it was way bigger change touching too many bases :) Choose your own adventure :)

Reply:
@vagababov in case he has some better options / opinions on this also before you go and make that change :)

Reply:
How about a new func with `port`?

Reply:
Send a PR in pkg: https://github.com/knative/pkg/pull/1006

Reply:
Used the latest change from `pkg/metrics`